pkg/vulnerability/vulnerability.go

```diff
@@ -149,7 +147,7 @@ func applyPolicy(ctx context.Context, vulns []types.DetectedVulnerability, polic
 		return nil, xerrors.Errorf("unable to prepare for eval: %w", err)
 	}
 
-	var filtered []types.DetectedVulnerability
+	filtered := make([]types.DetectedVulnerability, 0)
```

Comment:
I'm just curious which linter says this.

Reply:
https://stackoverflow.com/questions/59734706/how-to-resolve-consider-preallocating-prealloc-lint

Reply:
It is the case when we know the size in advance. In this case, we specify 0, so I think the performance is worse than before.

Reply:
<img width="1226" alt="Screenshot 2020-10-11 at 7 52 42 PM" src="https://user-images.githubusercontent.com/18293335/95681051-5672e800-0bfb-11eb-8792-8ffa5af6a563.png">

I don't see performance difference in the Benchmark test with 1000 size, do you want me to run with bigger numbers?

Reply:
BTW, I performance with preallocating the base capacity is better than not allocating
https://github.com/aquasecurity/trivy/pull/679/commits/179851c5c427bb6bdd187bac56ffb9ef35b3afd6
@knqyf263

Reply:
Ah, you didn't specify capacity. In that case, the performance should be the same as `var arr []int`. It means we don't have to change it. 

For example, `len(advisories)` is much greater than the length `vulns` requires. So, memory is wasted. 
https://github.com/aquasecurity/trivy/blob/179851c5c427bb6bdd187bac56ffb9ef35b3afd6/pkg/detector/library/advisory.go#L42

If the number of elements is the tens of thousands or millions and it takes a few minutes, it's better to specify the capacity beforehand. But in our case, it is only 10 or 100 and takes only in milliseconds. The performance may improve slightly, but it wastes memory (depending on the situation) and reduces readability. We don't have to think about performance for now and should keep it simple. I'm feeling like "premature optimization is the root of all evil".

Reply:
@knqyf263 Removed preallocate linter, reverted changes

Reply:
Also, I found a warning from golangci-lint.
https://github.com/golangci/golangci-lint/blob/796a95880598d8fd0a8b6b638d2419d84418e59c/.golangci.example.yml#L272